source/sixve.f90

```diff
@@ -211,397 +211,365 @@ subroutine envarsv(dpsv,oidpsv,rvv,ekv)
   use mod_alloc
 
   implicit none
-  integer ih1,ih2,j,kz1,l,l1,l2
-
-  !Local version of variables normally found in mod_common_main
-  real(kind=fPrec) aek,afok,as3,as4,as6,co,dpd,dpsq,dpsv,fi,    &
-        fok,fok1,fokqv,g,gl,hc,hi,hi1,hm,hp,hs,oidpsv,rho,rhoc,rhoi, &
-        rvv,si,siq,sm1,sm12,sm2,sm23,sm3,wf,wfa,wfhi
 
   real(kind=fPrec), allocatable, intent(inout) :: ekv(:,:) !(npart,nele)
 
-  dimension fokqv(npart),dpsv(npart)
-  dimension rvv(npart),oidpsv(npart)
-  dimension dpd(npart),dpsq(npart),fok(npart),rho(npart)
-  dimension fok1(npart),si(npart),co(npart),g(npart),gl(npart)
-  dimension sm1(npart),sm2(npart),sm3(npart),sm12(npart)
-  dimension as3(npart),as4(npart),as6(npart),sm23(npart)
-  dimension rhoc(npart),siq(npart),aek(npart),afok(npart)
-  dimension hp(npart),hm(npart),hc(npart),hs(npart),wf(npart)
-  dimension wfa(npart),wfhi(npart),rhoi(npart)
-  dimension hi(npart),fi(npart),hi1(npart)
+  integer ih1,ih2,j,kz1,l,l1,l2
 
-  real(kind=fPrec) fokm
+  ! Local version of variables normally found in mod_common_main
+  real(kind=fPrec) dpd(npart),dpsq(npart),dpsv(npart),fokqv(npart),oidpsv(npart),rvv(npart)
+  real(kind=fPrec) aek,afok,as3,as4,as6,co,fi,fok,fok1,g,gl,hc,hi,hi1,hm,hp,hs,rho,rhoc,rhoi,&
+    si,siq,sm1,sm12,sm2,sm23,sm3,wf,wfa,wfhi,fokm
 
-!-----------------------------------------------------------------------
   save
-!-----------------------------------------------------------------------
 
-  do 10 j=1,napx
-    dpd(j)=one+dpsv(j)
-    dpsq(j)=sqrt(dpd(j))
-10 continue
+  do j=1,napx
+    dpd(j)  = one+dpsv(j)
+    dpsq(j) = sqrt(dpd(j))
+  end do
+
   do 160 l=1,il
     do l1=1,6
       do j=1,napx
         do l2=1,2
-          al(l1,l2,j,l)=zero
-          as(l1,l2,j,l)=zero
+          al(l1,l2,j,l) = zero
+          as(l1,l2,j,l) = zero
         enddo
       enddo
     enddo
-    if(abs(el(l)).le.pieni) goto 160
-    kz1=kz(l)+1
-!       goto(20,40,80,60,40,60,100,100,140),kz1
-    if (kz1.eq.1) goto 20
-    if (kz1.eq.2) goto 40
-    if (kz1.eq.3) goto 80
-    if (kz1.eq.4) goto 60
-    if (kz1.eq.5) goto 40
-    if (kz1.eq.6) goto 60
-    if (kz1.eq.7) goto 100
-    if (kz1.eq.8) goto 100
-    if (kz1.eq.9) goto 140
+    if(abs(el(l)) <= pieni) goto 160
+
+    kz1 = kz(l)+1
+    if(kz1 == 1) goto 20
```

Comment:
Select case? Elseif?

Reply:
Can't because the goto labels are in use also aside from selecting on kz